base/compiler/abstractinterpretation.jl

```diff
@@ -1404,6 +1406,12 @@ function abstract_call_known(interp::AbstractInterpreter, @nospecialize(f),
             if !isvarargtype(aty)
                 ft = widenconst(aty)
                 if isa(ft, DataType) && isdefined(ft.name, :mt) && isdefined(ft.name.mt, :kwsorter)
+                    flag = get_curr_ssaflag(sv)
+                    if is_stmt_inline(flag)
+                        sv.kwfunc_inlining = true
+                    elseif is_stmt_noinline(flag)
+                        sv.kwfunc_inlining = false
+                    end
```

Comment:
this seems like some very incorrect global state changes for the containing method?

Reply:
yeah, somehow I assumed the containing method only has a single call of kwfunc at most, but it isn't true really.
The right way to support that is probably to propagate this information with a dedicated a lattice element.